otri/filtering/filters/generic_filter.py

```diff
@@ -1,42 +1,61 @@
-from ..filter import Filter, Stream, Collection
+from ..filter import Filter, Stream, Sequence, Mapping, Any
 from typing import Callable
 
 
 class GenericFilter(Filter):
     '''
-    Applies a given Callable to all input atoms and outputs them in the output Stream.
-    Inputs: a single Stream.
-    Outputs: a single Stream.
+    Applies a given Callable to all input data and outputs them in the output stream.
+    
+    Input:
+        Single stream.
+    Outputs:
+        Single stream.
     '''
 
-    def __init__(self, source_stream: Stream, operation: Callable):
+    def __init__(self, input: str, output: str, operation: Callable):
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Redefining built-in 'input'](https://app.codacy.com/gh/OTRI-Unipd/OTRI/pullRequest?prid=5649486)